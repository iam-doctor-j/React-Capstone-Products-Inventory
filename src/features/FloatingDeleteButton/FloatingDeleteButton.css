.circle {
    border-radius: 50%;
    box-shadow: 0 0.1rem 0.3rem rgba(0,0,0,0.5);
    padding: 1rem;
}
.floating-bottom {
    position: fixed;
    bottom: 10%;
    right: 10%;
    z-index: 10000;
}
.far {
    height: 0.5rem;
    width: 1.5rem;
}

/* @media only screen and (max-width: 550px) {
    .floating-bottom {
        left: 75%;
    }
}
@media only screen and (min-width: 550px) and (max-width: 850px) {
    .floating-bottom {
        left: 85%;
    }
} */
