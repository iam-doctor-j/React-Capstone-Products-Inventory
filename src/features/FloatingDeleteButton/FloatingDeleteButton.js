import React from "react";
import './FloatingDeleteButton.css'
import { deleteProductsFromDb } from "../ProductFeed/ProductFeedSlice";
import routes from "../../constants/routes";
import { toast } from "react-toastify";

export default function FloatingDeleteButton(props) {
    return(
        <button className="btn btn-danger circle floating-bottom" onClick={ () => {
            if(props.loggedIn) {
                console.log(props.deleteList)
                props.dispatch(deleteProductsFromDb(props.deleteList, () => toast.error('Product(s) deleted!')))
            } 
            else
                props.history.push(routes.LOGIN);
            } }>
            <i className="far fa-trash-alt"></i>
        </button>
    );
}