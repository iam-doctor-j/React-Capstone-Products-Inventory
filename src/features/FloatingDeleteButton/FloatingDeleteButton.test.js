import React from 'react';
import { shallow } from 'enzyme';
import FloatingDeleteButton from './FloatingDeleteButton';

describe('Test FloatingDeleteButton using shallow rendering', () => {
    let wrapper;

    const props = {
        loggedIn: true,
        deleteList: [],
        history: {
            push: () => {}
        }
    }

    beforeEach(() => {
        wrapper = shallow(<FloatingDeleteButton {...props}/>);
    });

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render one button', () => {
        expect(wrapper.find('button').length).toEqual(1);
    });

    afterEach(() => {
        wrapper.unmount();
    });
})