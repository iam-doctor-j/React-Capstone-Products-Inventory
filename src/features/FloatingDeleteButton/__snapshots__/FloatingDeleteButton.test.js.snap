// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FloatingDeleteButton using shallow rendering should render correctly 1`] = `
<button
  className="btn btn-danger circle floating-bottom"
  onClick={[Function]}
>
  <i
    className="far fa-trash-alt"
  />
</button>
`;
